logcheck/pkg/logcheck.go

```diff
@@ -496,3 +508,37 @@ func checkForIfEnabled(i *ast.IfStmt, pass *analysis.Pass, c *config) {
 			funcCall, varName, funcCall, varName, varName),
 	})
 }
+
+func checkForVerboseZero(expr *ast.CallExpr, pass *analysis.Pass) {
+	if typeAndValue, ok := pass.TypesInfo.Types[expr]; ok {
+		switch t := typeAndValue.Type.(type) {
+		case *types.Named:
+			if typeName := t.Obj(); typeName != nil {
+				if pkg := typeName.Pkg(); pkg != nil {
+					if typeName.Name() == "Verbose" && pkg.Path() == "k8s.io/klog/v2" {
+						lit, ok := expr.Args[0].(*ast.BasicLit)
+						if !ok || lit.Value != "0" {
+							return
+						}
+						msg := fmt.Sprintf("klog does not allow verbose zero")
+						pass.Report(analysis.Diagnostic{
+							Pos:     expr.Fun.Pos(),
+							Message: msg,
+						})
+					}
+					if typeName.Name() == "Logger" && pkg.Path() == "github.com/go-logr/logr" {
```

Comment:
This and the if check for klog above seem to match on the result of the function call, right? While that works for the current klog API (only `V()` returns a `Verbose`) it's probably wrong for logger, because it will also match for e.g. `logger.WithValues`.

I think in both cases it should check for a function call that invokes `klog.V` or `Logger.V`.

The only difference between the two is the message string. I think we can use the same code for both packages and just say something about `V`, which can be the same for both.